R/expr-string.R

```diff
@@ -0,0 +1,1158 @@
+#' df$select(pl$col("foo")$str$strip_chars(" hel rld"))
+expr_str_strip_chars <- function(matches = NULL) {
+  self$`_rexpr`$str_strip_chars(as_polars_expr(matches, as_lit = TRUE)$`_rexpr`) |>
+    wrap()
+}
+
+#' Strip leading characters
+#'
+#'
+#' @aliases expr_str_strip_chars_start
+#' @description  Remove leading characters.
+#'
+#' @param matches The set of characters to be removed. All combinations of this
+#' set of characters will be stripped. If `NULL` (default), all whitespace is
+#' removed instead. This can be an Expr.
+#'
+#' @details
+#' This function will not strip any chars beyond the first char not matched.
+#' `strip_chars_start()` removes characters at the beginning of the string only.
+#' Use `strip_chars()` and `strip_chars_end()` to remove characters from the left
+#' and right or only from the right respectively.
+#' @return Expr of String lowercase chars
+#' @examples
+#' df <- pl$DataFrame(foo = c(" hello", "\tworld"))
+#' df$select(pl$col("foo")$str$strip_chars_start(" hel rld"))
+expr_str_strip_chars_start <- function(matches = NULL) {
+  self$`_rexpr`$str_strip_chars_start(as_polars_expr(matches, as_lit = TRUE)$`_rexpr`) |>
+    wrap()
+}
+
+#' Strip trailing characters
+#'
+#'
+#' @aliases expr_str_strip_chars_end
+#' @description  Remove trailing characters.
+#'
+#' @param matches The set of characters to be removed. All combinations of this
+#' set of characters will be stripped. If `NULL` (default), all whitespace is
+#' removed instead. This can be an Expr.
+#'
+#' @details
+#' This function will not strip any chars beyond the first char not matched.
+#' `strip_chars_end()` removes characters at the end of the string only.
+#' Use `strip_chars()` and `strip_chars_start()` to remove characters from the left
+#' and right or only from the left respectively.
+#' @return Expr of String lowercase chars
+#' @examples
+#' df <- pl$DataFrame(foo = c(" hello", "\tworld"))
+#' df$select(pl$col("foo")$str$strip_chars_end(" hel\trld"))
+#' df$select(pl$col("foo")$str$strip_chars_end("rldhel\t "))
+expr_str_strip_chars_end <- function(matches = NULL) {
+  self$`_rexpr`$str_strip_chars_end(as_polars_expr(matches, as_lit = TRUE)$`_rexpr`) |>
+    wrap()
+}
+
+
+#' Fills the string with zeroes.
+#'
+#' @aliases expr_str_zfill
+#' @description Add zeroes to a string until it reaches `n` characters. If the
+#' number of characters is already greater than `n`, the string is not modified.
+#' @keywords ExprStr
+#' @param alignment Fill the value up to this length. This can be an Expr or
+#' something coercible to an Expr. Strings are parsed as column names.
+#' @details
+#' Return a copy of the string left filled with ASCII '0' digits to make a string
+#' of length width.
+#'
+#' A leading sign prefix ('+'/'-') is handled by inserting the padding after the
+#' sign character rather than before. The original string is returned if width is
+#' less than or equal to `len(s)`.
+#'
+#' @return Expr
```

Comment:
Please inherit return from `as_polars_expr`

Reply:
Should this always be the case? Sometimes we specify the datatype of the Expr output, e.g. "Expr of Date, Datetime or Time type". This is useful information I think.

Reply:
I understand your point, but I think there will be less friction if we stick to Python Polars clones.